pytest_fxa/plugin.py

```diff
@@ -78,16 +93,11 @@ def fxa_account(fxa_client, fxa_email):
                                         fxa_account.password)
     logger.info('Created: {}'.format(fxa_account))
     account.fetch()
-    message = account.wait_for_email(lambda m: 'x-verify-code' in m['headers'])
+    message = account.wait_for_email(
+        lambda m: 'x-verify-code' in m['headers'] and
+        session.uid == m['headers']['x-uid']
+    )
     session.verify_email_code(message['headers']['x-verify-code'])
     logger.info('Verified: {}'.format(fxa_account))
```

Comment:
We'll also want to call `fxa_cleanup` in the event that the fetch/wait/verify fails. Perhaps a try/finally from line 80. In fact, if the yield is within the try, then the finally would be the only cleanup we need as it should execute on success and failure. It would still be worth adding a test for this.

Reply:
Do we want to raise an error there?

Reply:
I don't understand the question. We'd want any errors to be raised except for an attempt to destroy/cleanup an account that doesn't exist.